### ISSUE TITLE
'13' as invalid search string

### ISSUE THREAD

**Issue by aschwanb** [ninxsoft/mist-cli#131]
Mist does fail to download a valid installer if only the main OS version is give. I'm not sure if this is intentional or not. However, it used to work and it currently breaks other tools (like erase-install) which expect this to work.

## Version
```zsh
% mist --version
1.11 (Latest: 1.11)
```

## Command
```zsh
sudo mist download installer 13 application --application-name "Install %NAME%.app" --output-directory /Applications --no-ansi  
```

## Expected
mist does download the latest version of macOS 13. (13.3.1 in this case)

## Actual
mist does fail with `No macOS Installer found with '13', exiting`

```zsh
% sudo mist download installer 13 application --application-name "Install %NAME%.app" --output-directory /Applications --no-ansi  
Password:
┌──────────────────┐
│ INPUT VALIDATION │
└──────────────────┘
  ├─ User is 'root'...
  ├─ Download search string will be '13'...
  ├─ Include betas in search results will be 'false'...
  ├─ Only include compatible installers will be 'false'...
  ├─ Cache downloads will be 'false'...
  ├─ Output directory will be '/Applications'...
  ├─ Temporary directory will be '/private/tmp/com.ninxsoft.mist'...
  ├─ Force flag has not been set, existing files will not be overwritten...
  ├─ Application name will be 'Install %NAME%.app'...
┌────────┐
│ SEARCH │
└────────┘
  ├─ Searching for macOS download '13'...
  └─ No macOS Installer found with '13', exiting...
```

**Comment by ninxsoft** [comment]
@aschwanb Nice catch - looks like this bug was introduced when we attempted to filter on exact matches only: #122

Investigating a way forward that strike a happy balance between exact / fuzzy matching, stay tuned 👍

**Comment by ninxsoft** [comment]
In the interim, can you move forward with the following?

```bash
sudo mist download installer ventura application --application-name "Install %NAME%.app" --output-directory /Applications --no-ansi
```

**Comment by grahampugh** [comment]
I don't think that macOS 13 is ever a specific version - it's 13.0. So the logic could probably see if there's any dot in the search string and if so, treat it as an exact search, whereas if not, go for the latest available.

**Comment by aschwanb** [comment]
> In the interim, can you move forward with the following?
> 
> ```shell
> sudo mist download installer ventura application --application-name "Install %NAME%.app" --output-directory /Applications --no-ansi
> ```

For the record: This works as intended:
`Found [032-66588] macOS Ventura 13.3.1 (22E261) [2023-04-17]`

**Comment by ninxsoft** [comment]
@aschwanb + @grahampugh looks like the fix was more trivial than expected: #132

- `mist download firmware "Ventura"` will download the latest (eg. macOS Ventura 13.4 22F66)
- `mist download firmware "13"` will download the latest (eg. macOS Ventura 13.4 22F66)
- `mist download firmware "13.3.1"` will download a specific version (eg. macOS Ventura 13.3.1 22E261)
- `mist download firmware "13.0"` will download a specific version (eg. macOS Ventura 13.0 22A380)
